File: src/types/file.ts
export interface CloudFile {
  id: number;
  code: string;
  file_path: string;
  file_name: string | null;
  file_size: number | null;
  provider: string;
  provider_file_id: string | null;
  provider_url: string | null;
  provider_meta: string | null;
}

export interface ProviderFileEntry {
  file_id: string;
  file_name: string;
  file_size: number | null;
  file_url: string | null;
  is_directory: boolean;
}

export interface ProviderConfigData {
  id: string;
  name: string;
  provider_type: string; // "local" | "webdav" | "open115"
  endpoint: string | null;
  username: string | null;
  password: string | null;
  root: string | null;
}

File: src-tauri/src/models/file.rs
use serde::{Deserialize, Serialize};

#[allow(dead_code)]
#[derive(Debug, Serialize, Deserialize, Clone)]
pub struct CloudFile {
    pub id: i64,
    pub code: String,
    pub file_path: String,
    pub file_name: Option<String>,
    pub file_size: Option<i64>,
    pub provider: String,
    pub provider_file_id: Option<String>,
    pub provider_url: Option<String>,
    pub provider_meta: Option<String>,
}

/// File entry returned by a provider's list/search operation.
/// Not persisted directly — used as an intermediate representation
/// before the user decides to link a file to a movie.
#[derive(Debug, Serialize, Deserialize, Clone)]
pub struct ProviderFileEntry {
    pub file_id: String,
    pub file_name: String,
    pub file_size: Option<i64>,
    pub file_url: Option<String>,
    pub is_directory: bool,
}
